flair/datasets/entity_linking.py

```diff
@@ -1760,3 +2210,398 @@ def __init__(
+                    assert end not in start_ids
+                    end_ids.append(end)
+                    start_ids.append(end)
+        start_ids.sort()
+        end_ids.sort()
+        passage_sentences = []
+        n_tokens = len(start_ids)
+        for passage in entry["passages"]:
+            token_offset = passage["offsets"][0][0]
+            start_idx = bisect.bisect_left(start_ids, token_offset)
+            end_idx = bisect.bisect_right(end_ids, passage["offsets"][0][1])
+            offsets = zip(start_ids[start_idx:end_idx], end_ids[start_idx:end_idx])
+            passage_tokens = [
+                Token(passage["text"][0][start - token_offset : end - token_offset]) for start, end in offsets
+            ]
+            for i, idx in enumerate(range(start_idx, end_idx)):
+                if idx + 1 < n_tokens:
+                    passage_tokens[i].whitespace_after = start_ids[idx + 1] - end_ids[idx]
+            passage_sentences.append(Sentence(passage_tokens, start_position=token_offset))
+            for token, start, end in zip(
+                passage_sentences[-1], start_ids[start_idx:end_idx], end_ids[start_idx:end_idx]
+            ):
+                assert token.start_position + passage_sentences[-1].start_position == start
+                assert token.end_position + passage_sentences[-1].start_position == end
+
+        start_id_to_token = {
+            token.start_position + sentence.start_position: (sentence, i)
+            for sentence in passage_sentences
+            for i, token in enumerate(sentence)
+        }
+        end_id_to_token = {
+            token.end_position + sentence.start_position: (sentence, i)
+            for sentence in passage_sentences
+            for i, token in enumerate(sentence)
+        }
+        for entity in entities:
+            mention_ids = [":".join([n[key] for key in self.norm_keys]) for n in entity["normalized"]]
+            assert len(entity["offsets"]) == len(entity["text"])
+            for (start, end), text in zip(entity["offsets"], entity["text"]):
+                assert start in start_id_to_token
+                assert end in end_id_to_token
+                sent_s, start_token_idx = start_id_to_token[start]
+                sent_e, end_token_idx = end_id_to_token[end]
+                assert sent_s is sent_e
+
+                for mention_id in mention_ids:
+                    sent_s[start_token_idx : end_token_idx + 1].add_label(self.label_type, mention_id)
+        return passage_sentences
+
+    def _files_to_dataset(self, paths: Union[Path, List[Path]]) -> FlairDatapointDataset:
+        if isinstance(paths, Path):
+            paths = [paths]
+        all_sentences = []
+        for path in paths:
+            for entry in self._file_to_dicts(path):
+                all_sentences.extend(self._dict_to_sentences(entry))
+        return FlairDatapointDataset(all_sentences)
+
+
+class BIGBIO_NCBI_DISEASE(BigbioCorpus):
```

Comment:
Also for each of the BigbioCorpus subclasses please add a brief description of the dataset and link.